misc/LADYBUG_USER_STORIES.md

```diff
@@ -3,51 +3,152 @@ Ladybug User Stories
+
+**2020:** Given is that I am viewing a report as a Frank tester. I want the option to declare some checkpoints **stubbed**, checkpoints that correspond to calls to external systems. When the report is rerun, the Frank!Framework should not call the external systems again but it should return the results already stored in the stubbed checkpoints. This way, only the logic within the Frank configuration captured in the report is tested, not the behavior of the external system. Stubbing allows Frank testers to work with a simpler test environment, because the test do not require access to external systems.
+
+**2030:** Given is that I am viewing a report as a Frank tester. I want the option to base parameterized tests upon my report. This means that I introduce variable references in my report. I can create a new report by specifying values for the variables. I can do this for multiple possibilities to set the variables, all resulting in a new clone of the report.
+
+**2040:** Given is that I am viewing a report as a Frank tester. I want the option to add a *description* to my report. In the description I can document what the report viewed as a test case should test.
+
+# I want to re-run (test) reports to test my Frank application
+
+**3000:** As a Frank tester, when I enter the test tab I want to see all reports that I have prepared as test cases for my Frank application.
+
+**3010:** As a Frank tester, I want the option to *rerun* reports. Rerunning a report means that the Frank adapter that produced the report is re-executed. The same input message is supplied. For each pipe, the produced output is compared with the value stored with the corresponding checkpoint. The test succeeds if the new messages are the same after applying the configured XSLT transformation (user story **2010**).
+
+**3020:** As a Frank tester, I want the option to rerun a single report. I should have that option for every report, whether it is in the Debug tab or in the Test tab. This way, I am not obliged to edit reports before they can act as automated tests.
+
+**3030:** As a Frank tester, I want the option to organize the reports in the Test tab. I want to create groups of tests that can have sub-groups. A tree structure in which the composite nodes are test groups and the leaf nodes are reports. This structure gives me an overview of the tests I have.
+
+**3040:** As a Frank tester, I want the option to select test cases and groups of test cases. I want the option to rerun the tests I selected. This gives me fine-grained control over my test runs when I want to test parts of my Frank application.
```

Comment:
De eerste twee zinnen zijn aparte user stories.

Reply:
Wat is het nut van selecteren als je niets kunt doen met wat je geselecteerd hebt? Daarom was dit één user story.